core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/rabbitmq.yml

```diff
@@ -1,30 +1,57 @@
 ---
-- name: RabbitMQ | Check version
-  command: rabbitmqctl version
-  register: rabbitmq_version_current
+- name: RabbitMQ | Register version output
+  command: rabbitmq-diagnostics server_version
+  register: rabbitmq_version
   changed_when: false
 
+- name: RabbitMQ | Register cluster status output
+  command: rabbitmqctl cluster_status --formatter json
+  register: rabbitmq_cluster_status
+  changed_when: false
+  run_once: true
+
 - name: RabbitMQ | Include defaults from rabbitmq role
   include_vars:
     file: roles/rabbitmq/defaults/main.yml
 
-- name: RabbitMQ | Set rabbitmq_master fact
+- name: RabbitMQ | Set facts
   set_fact:
     rabbitmq_master: "{{ groups['rabbitmq'][0] }}"
```

Comment:
Is it an assumption or a fact? Rabbitmq uses the Raft protocol can we safely assume 0-th node to be a master here? :thinking:

Reply:
The answer is [here](https://www.rabbitmq.com/upgrade.html#full-stop-upgrades):

```
During an upgrade, the last disc node to go down must be the first node to be brought online. Otherwise the started node will emit an error message and fail to start up. Unlike an ordinary cluster restart, upgrading nodes will not wait for the last disc node to come back online.

While not strictly necessary, it is a good idea to decide ahead of time which disc node will be the upgrader, stop that node last, and start it first. Otherwise changes to the cluster configuration that were made between the upgrader node stopping and the last node stopping will be lost.
```

So here it's just a decision to mark the "master"  node as an "upgrader".